nats.go

```diff
@@ -2431,8 +2431,15 @@ func (nc *Conn) processInfo(info string) error {
 		}
 		nc.addURLToPool(fmt.Sprintf("%s://%s", nc.connScheme(), curl), true, saveTLS)
 	}
-	if hasNew && !nc.initc && nc.Opts.DiscoveredServersCB != nil {
-		nc.ach.push(func() { nc.Opts.DiscoveredServersCB(nc) })
+	if hasNew {
+		// If randomization is allowed, randomize the pool now that we have
+		// added all new URLs.
+		if !nc.Opts.NoRandomize {
+			nc.shufflePool()
```

Comment:
Should we shuffle with the new one added?

Reply:
That's what I do. Above we add all new ones (as long as they are new). The `hasNew` boolean will be true if at least 1 URL was added. If that is the case then we shuffle the whole pool.